cf-testd/cf-testd.c

```diff
@@ -471,12 +542,49 @@ int main(int argc, char *argv[])
     if (config->file != NULL)
     {
         Log(LOG_LEVEL_NOTICE, "Got file argument: '%s'", config->file);
-        Log(LOG_LEVEL_ERR, "File input not yet supported");
-        // TODO: implement file input
-        exit(EXIT_FAILURE);
+        FILE *fp = NULL;
+        char buffer[CF_BUFSIZE];
+        int c, i = 0;
+
+        if ((fp = fopen(config->file, "r")) == NULL)
+        {
+            Log(LOG_LEVEL_INFO,
+                "Could not read report file at %s (fopen: %s)",
+                config->file,
+                GetErrorStr());
+            exit(EXIT_FAILURE);
+        }
+
+        while ((c = getc(fp)) != EOF)
+        {
+            if (i > CF_BUFSIZE)
+            {
+                Log(LOG_LEVEL_ERR,
+                    "Report file was larger than max buffer size %d",
+                    CF_BUFSIZE);
+                exit(EXIT_FAILURE);
+            }
+            buffer[i++] = c;
+        }
+        Log(LOG_LEVEL_NOTICE, "Read %d bytes for report contents", i);
+        buffer[i] = 0;
+
+        if (i == 0)
+        {
+            Log(LOG_LEVEL_ERR, "Report file contained no bytes");
+            exit(EXIT_FAILURE);
+        }
+
+        config->report     = buffer;
+        config->report_len = i - 1;
```

Comment:
Just to be 100% clear, these derefs cannot be removed, since these are assignments. My comment above is about the usage of `config->report` and `config->report_len` in arguments etc.